environment/src/main/java/com/sequenceiq/environment/environment/poller/DatahubPollerProvider.java

```diff
@@ -140,4 +167,54 @@ private boolean stackAndClusterStopped(StackV4Response stackV4Response, ClusterV
                 && cluster.getStatus() != null
                 && cluster.getStatus().isStopped();
     }
+
+    private List<AttemptResult<Void>> collectUpgradeCcmResults(List<FlowIdentifier> mutableFlowIds, List<FlowIdentifier> remainingFlowIds, Long envId) {
+        if (PollGroup.CANCELLED.equals(EnvironmentInMemoryStateStore.get(envId))) {
+            String message = "Datahub polling cancelled in inmemory store, id: " + envId;
+            LOGGER.info(message);
+            throw new PollerStoppedException(message);
+        }
+
+        return mutableFlowIds.stream()
+                .map(flowId -> fetchUpgradeCcmResult(remainingFlowIds, flowId))
+                .collect(Collectors.toList());
+    }
+
+    private AttemptResult<Void> fetchUpgradeCcmResult(List<FlowIdentifier> remainingFlowIds, FlowIdentifier flowId) {
+        FlowCheckResponse flowCheckResponse;
+        switch (flowId.getType()) {
```

Comment:
Can we have a debug log about which flow is checked by it's id?

Reply:
sure